### ISSUE TITLE
The position of `dropout` operation are different from official repo ?

### ISSUE THREAD

**Issue by zerlinwang** [nikhilbarhate99/min-decision-transformer#2]
In official repo, Attention part:
`       

        w = nn.Softmax(dim=-1)(w)
        w = self.attn_dropout(w)
        # Mask heads if we want to
        if head_mask is not None:
            w = w * head_mask

        outputs = [torch.matmul(w, v)]`
The `dropout` is directly after the `Softmax` and before the `matmul`.

On the other hand, in our implement:
`     
        
        normalized_weights = F.softmax(weights, dim=-1)
        # attention (B, N, T, D)
        # normalized_weights.shape: (B, N, T, T)
        # v.shape: (B, N, T, D)
        attention = self.att_drop(normalized_weights @ v)`
The `dropout` is at last.

In my opinion, they are different. How do you think about it? :-)

**Comment by nikhilbarhate99** [comment]
Yes, this is not an exact reproduction of the Decision Transformer paper.

So some differences and simplifications are expected as mentioned in the introduction.

In this case though, I do not think it will make much difference, because dropout randomly zeros the input, so zeroing the weights and then multiplying with the values is the same as zeroing the final attention values.

**Comment by zerlinwang** [comment]
> Yes, this is not an exact reproduction of the Decision Transformer paper.
> 
> So some differences and simplifications are expected as mentioned in the introduction.
> 
> In this case though, I do not think it will make much difference, because dropout randomly zeros the input, so zeroing the weights and then multiplying with the values is the same as zeroing the final attention values.

Thank you for your reply and excellent reproduction! It helps a lot.
